src/java.base/share/classes/javax/crypto/KEM.java

```diff
@@ -0,0 +1,613 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package javax.crypto;
+
+import sun.security.jca.GetInstance;
+
+import java.security.*;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.spec.AlgorithmParameterSpec;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * The Key Encapsulation Mechanism.
+ * <p>
+ * The {@code getInstance} method creates a new {@code KEM} object that
+ * implements the specified algorithm.
+ * <p>
+ * A {@code KEM} object is immutable. It is safe to call multiple
+ * {@code newEncapsulator} and {@code newDecapsulator} methods on the
+ * same {@code KEM} object at the same time.
+ * <p>
+ * If no provider is specified in the {@code getInstance} method when
+ * instantiating a {@code KEM} object, depending on the type of private key
+ * or public key, and the optional {@code AlgorithmParameterSpec} object,
+ * the {@code newEncapsulator} or {@code newDecapsulator} method
+ * may return encapsulators or decapsulators from different providers. The user
```

Comment:
This sentence is kind of long. Suggest breaking it up, ex: "If a provider is not specified in the {@code getInstance} method when  instantiating a {@code KEM} object,  the {@code newEncapsulator} and {@code newDecapsulator} method may return encapsulators or decapsulators from different providers. The provider selected is based on the parameters passed to the the {@code newEncapsulator} or {@code newDecapsulator} methods: the private or public key and the optional {@code AlgorithmParameterSpec}."